cgp-utils_config.md

```diff
@@ -0,0 +1,114 @@
+# Configuration
+
+The [cpg-utils library](https://github.com/populationgenomics/cpg-utils) ([pypi](https://pypi.org/project/cpg-utils/)) contains a streamlined config management tool. This config management is used by most production CPG workflows, but is useful in projects or scripts at any scale.
+
+This configuration tool uses one or more `TOML` files, and creates a dictionary of key-value attributes which can be accessed at any point, without explicitly passing a configuration object.
+
+## TOML
+
+[Tom's Obvious, Minimal Language](https://toml.io/en/) is a config file format designed to be easily human readable and writeable, with clear data structures. Sections are delineated using bracketed headings, and key-value pairs are defined using `=` syntax, e.g. :
+
+```toml
+global_key = "value"
+
+[heading_1]
+name = "Luke Skywalker"
+age = 53
+
+[heading_1.subheading]
+occupation = ["Jedi", "Hermit", "Force Ghost"]
+```
+
+will be digested into the simple dictionary:
+
+```python
+{
+    'global_key': 'value',
+    'heading_1': {
+        'name': 'Luke Skywalker',
+        'age': 53,
+        'subheading': {
+            'occupation': ["Jedi", "Hermit", "Force Ghost"]
+        }
+    }
+}
+```
+
+## Using cpg-utils.config
+
+To use the `cpg-utils.config` functions, import these two key methods into any script:
+
+```python
+from cpg_utils.config import get_config, set_config_paths
+```
+
+Configuration is loaded lazily, so config files are loaded into memory only when we try to retrieve values from them. This reduces start-up overhead for scripts using this config approach, but can result in late failures if files with invalid content are specified.
+
+### Setting configuration content
+
+Configuration files can be set in two ways:
+
+1. Set the `CPG_CONFIG_PATH` environment variable
+2. Use `set_config_paths` to point to one or more config TOMLs
+
+`CPG_CONFIG_PATH` can be a comma-delimited String, and the argument to `set_config_paths` is a list of `strings`. The
+config functionality allows for multiple files to be specified, applied in the following way:
+
+1. The cpg-utils repository contains a template config,
```

Comment:
Also worth mentioning that the template contains descriptions of the parameters as comments. If anybody adds a generically useful config parameter, the template "documentation" should also be updated.